serverless/aws/lambda/src/main/scala/sttp/tapir/serverless/aws/lambda/AwsToResponseBody.scala

```diff
@@ -0,0 +1,55 @@
+package sttp.tapir.serverless.aws.lambda
+
+import sttp.capabilities
+import sttp.model.HasHeaders
+import sttp.tapir.internal.NoStreams
+import sttp.tapir.server.interpreter.ToResponseBody
+import sttp.tapir.{CodecFormat, RawBodyType, WebSocketBodyOutput}
+
+import java.io.InputStream
+import java.nio.ByteBuffer
+import java.nio.charset.Charset
+import java.util.Base64
+
+private[lambda] class AwsToResponseBody[F[_]](implicit options: AwsServerOptions[F]) extends ToResponseBody[String, Nothing] {
+  override val streams: capabilities.Streams[Nothing] = NoStreams
+
+  override def fromRawValue[R](v: R, headers: HasHeaders, format: CodecFormat, bodyType: RawBodyType[R]): String = bodyType match {
+    case RawBodyType.StringBody(charset) =>
+      val str = v.asInstanceOf[String]
+      if (options.encodeResponseBody) Base64.getEncoder.encodeToString(str.getBytes(charset)) else str
+
+    case RawBodyType.ByteArrayBody =>
+      val bytes = v.asInstanceOf[Array[Byte]]
+      if (options.encodeResponseBody) Base64.getEncoder.encodeToString(bytes) else new String(bytes)
+
+    case RawBodyType.ByteBufferBody =>
+      val byteBuffer = v.asInstanceOf[ByteBuffer]
+      if (options.encodeResponseBody) Base64.getEncoder.encodeToString(safeRead(byteBuffer)) else new String(safeRead(byteBuffer))
+
+    case RawBodyType.InputStreamBody =>
+      val stream = v.asInstanceOf[InputStream]
+      if (options.encodeResponseBody) Base64.getEncoder.encodeToString(stream.readAllBytes()) else new String(stream.readAllBytes())
+
+    case RawBodyType.FileBody         => throw new UnsupportedOperationException
+    case _: RawBodyType.MultipartBody => throw new UnsupportedOperationException
+  }
+
+  private def safeRead(byteBuffer: ByteBuffer): Array[Byte] = {
+    if (byteBuffer.hasArray) {
+      byteBuffer.array()
```

Comment:
I think the array can be larger than the actual data. See: https://github.com/softwaremill/sttp/commit/d1e3ec33df0b902229d9891cd0e785f7c95caf3d

We'd have to check the array size, the buffer's limit and if they don't match, create a copy

Follow-up task (not in this PR): check other interpreters if they don't have the same bug

Reply:
I wrote it down